pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -700,6 +700,12 @@ class Timestamp(_Timestamp):
         """
         return self.tzinfo
 
+    @tz.setter
+    def tz(self, value):
+        # GH 3746
```

Comment:
Can you put here a more informative comment instead of referring to the issue (if it is needed of course)? 
I think the only reason to have this is to have a more informative error message?